### ISSUE TITLE
MutatingLdpHandler::updateMemento isn't always needed

### ISSUE THREAD

**Issue by ajs6f** [trellis-ldp/trellis#325]
`MutatingLdpHandler::updateMemento` gets called by mutating request methods in `TrellisHttpResouce`, but some backends (e.g. `trellis-cassandra`) don't require it because they store time series (version everything all the time) and to make a mutation in those backends is to store a Memento of it. Calling it when not needed results in at least one unnecessary `ResourceService::get`, which touches persistence.

It should be possible somehow to avoid this inefficiency, although I admit it could be tricky to avoid backend knowledge of HTTP mechanics.

**Comment by ajs6f** [comment]
Well, that's better than what I have now, which only NOOPs the `put` that is there now. I assume `-http` would call this? It's weird in the sense that why is anyone calling one service with another as param. But maybe that's okay?

**Comment by acoburn** [comment]
Yes, this would be called in the `-http` layer (basically, replacing [this code](https://github.com/trellis-ldp/trellis/blob/master/core/http/src/main/java/org/trellisldp/http/impl/MutatingLdpHandler.java#L135-L136)).
